packages/runtimes/js/src/bundle-url.js

```diff
@@ -1,28 +1,22 @@
-/* globals document:readonly */
-
-var bundleURL = null;
-function getBundleURLCached() {
-  if (!bundleURL) {
-    bundleURL = getBundleURL();
+var bundleURL = {};
+function getBundleURLCached(id) {
+  var value = bundleURL[id];
+  if (!value) {
+    value = getBundleURL();
+    bundleURL[id] = value;
   }
 
-  return bundleURL;
+  return value;
 }
 
 function getBundleURL() {
-  // Attempt to find the URL of the current script and use that as the base URL
-  if (process.env.PARCEL_BUILD_ENV === 'test') {
-    if (typeof document !== 'undefined' && 'currentScript' in document) {
-      return getBaseURL(document.currentScript.src);
-    }
-  }
-
   try {
     throw new Error();
   } catch (err) {
     var matches = ('' + err.stack).match(/(https?|file|ftp):\/\/[^)\n]+/g);
     if (matches) {
-      return getBaseURL(matches[0]);
+      // Use the last occurrence so that the URL of the calling bundle is returned, not of the bundling containing this
+      return getBaseURL(matches[matches.length - 1]);
```

Comment:
Couldn't one bundle call another bundle which then calls getBundleUrl and then the last item in the stack would be from the first bundle rather than the bundle that called getBundleUrl?